py3status/modules/usbguard.py

```diff
@@ -0,0 +1,178 @@
+                self.parent.device = device
+        self.parent.py3.update()
+
+    # on policy change signal
+    def _cb_devices_policy_changed(self, *args):
+        device = self.parent.device
+        device_args = args[4]
+        device_perms = device_args[3]
+        actions = ['allow', 'reject']
+        for x in actions:
+            if x in device_perms:
+                device['device_name'] = None
+                device['device_id'] = None
+                device['device_hash'] = None
+                self.parent.device = device
+                self.parent.py3.update()
+                break
+
+    def run(self):
+        while not self.parent.killed.is_set():
+            self._setup_dbus()
+            self.dbus.subscribe(
+                object=self.dbus_devices_filter,
+                signal='DevicePresenceChanged',
+                signal_fired=self._cb_devices_presence_changed)
+
+            self.dbus.subscribe(
+                object=self.dbus_devices_filter,
+                signal='DevicePolicyChanged',
+                signal_fired=self._cb_devices_policy_changed)
+
+            self.loop = GLib.MainLoop()
+            self.loop.run()
+
+
+class Py3status:
+    # available configuration options
+    allow_urgent = False
+    button_allow = 1
+    button_block = None
+    button_reject = 3
+    format = u'[UsbGuard: [{device_name}]]'
+
+    def post_config_hook(self):
+        self.device = {
+            'device_name': None,
+            'device_id': None,
+            'device_hash': None
+        }
+        self.error = None
+        self.dbus = SystemBus()
+        try:
+            self.proxy = self.dbus.get(DBUS_INTERFACE)
+        except:
+            self.parent.error = Exception("usbguard-dbus service not running")
+
+        self.targets = {'allow': 0, 'block': 1, 'reject': 2}
+        self.killed = threading.Event()
+        UsbguardListener(self).start()
+
+    def _usbguard_dbus_cmd(self, action):
+        return self.proxy.applyDevicePolicy(self.device['device_id'],
+                                            self.targets[action], 0)
+
+    def usbguard(self):
+        """
+        """
+        if self.error:
+            self.py3.error(str(self.error), self.py3.CACHE_FOREVER)
+
+        response = {
+            'cached_until': self.py3.CACHE_FOREVER,
+            'full_text': self.py3.safe_format(self.format, self.device)
+        }
+        if self.allow_urgent:
+            response['urgent'] = True
+        return response
+
+    def on_click(self, event):
+        """
+        """
+        button = event['button']
+        action = None
+        if self.button_allow and button == self.button_allow:
+            action = 'allow'
+        elif self.button_reject and button == self.button_reject:
+            action = 'reject'
+        elif self.button_block and button == self.button_block:
+            action = 'block'
+
+        if action:
+            self._usbguard_dbus_cmd(action)
```

Comment:
(Opinion) I think it is possible to expand this more if users can specify a list of buttons to toggle (via a single button) similar to `xrandr` instead of individual buttons.  Do I want `reject` or `block`... or both?

Specify it via...
* `actions = ['allow', 'block']`
* `actions = ['allow', 'reject']`
* `actions = ['allow', 'block', 'reject']`
* `actions = ['allow', 'reject', 'block']`

Assuming users have `allow_urgent = False`, we could add `thresholds` to print color states so each device (if you implement this) could have different color state..... for fun or something.

Reply:
Why not for the first proposal.

But second, I see the idea but afraid this can become very heavy, yet main goal at my eyes is not to keep device display in the bar, display all usb devices plugged will be a bad idea, i think anyone will do this or maybe with a 50" monitor :P

Reply:
Your module, your call. We're having fun here. :-)
>display all usb devices plugged will be a bad idea

Your module works fine for me. I looked at your code. Seems okay too athough I think you should get rid of `device_` prefix.. and to pass everything in, not just 3 placeholders.

However, it could be neat to toggle on/off usb devices. Got a webcam? Keep that disabled until you need it... Same for touchpad+onboard keyboard (if you're using mech keyboard), and whatever else. I have Xbox360 controller and I'd want to keep that off instead of unplugging it because it rumbles every time the system wakes up... or something.

Maybe we can add `devices = []` filter too.. so not everything will be on the bar, only what you want it to be on the bar. Although users can always use `usbguard-applet-qt`. I think we can use `frame` module or build our own `frame` to deal with too many devices. Not sure. Many things are possible.

Btw, what (exactly) did you not like about `usbguard-applet-qt`?

Reply:
Will think of this, good idea for toggle webcam ;)

The ugly one (but functional)
![image](https://user-images.githubusercontent.com/519083/41932945-c6c684b2-7982-11e8-8b1d-0f05c4c51bab.png)

Reply:
Is it a matter of theming issue? I don't have gaps between texts and it does not look horrible here. 

![2018-06-26-194551_usbguard-qt](https://user-images.githubusercontent.com/852504/41946894-2c3fbb14-797a-11e8-9858-41b6b3341fc6.png)

Reply:
What is the game beside? An emulator?

Reply:
https://www.twitch.tv/gamesdonequick for few days.